server/http.go

```diff
@@ -525,18 +535,32 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 	userPMISettingPref, err := p.getPMISettingData(user.Id)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
-		p.askPreferenceForMeeting(user.Id, req.ChannelID)
+		p.askPreferenceForMeeting(user.Id, req.ChannelID, req.RootID)
 		return
 	}
 
 	createMeetingWithPMI := false
 	switch userPMISettingPref {
 	case zoomPMISettingValueAsk:
-		p.askPreferenceForMeeting(user.Id, req.ChannelID)
+		p.askPreferenceForMeeting(user.Id, req.ChannelID, req.RootID)
+
+		if _, err = w.Write([]byte(`{"meeting_url": ""}`)); err != nil {
+			p.API.LogWarn("failed to write the response", "Error", err.Error())
+		}
 		return
 	case "", trueString:
 		createMeetingWithPMI = true
 		meetingID = zoomUser.Pmi
+
+		if meetingID <= 0 {
+			p.sendEnableZoomPMISettingMessage(userID, req.ChannelID, req.RootID)
+			meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
+			if createMeetingErr != nil {
+				p.API.LogWarn("failed to create the meeting", "Error", createMeetingErr.Error())
+				http.Error(w, createMeetingErr.Error(), http.StatusInternalServerError)
```

Comment:
Does this call to `http.Error` make this error show in the client? Maybe we want something more user-friendly if so. Wondering about the other calls to `http.Error` as well

Reply:
It depends how low-level or cryptic these errors are. I trust your judgment on this

Reply:
@mickmister Updated, displaying a generic error to the user now.